compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1580,9 +1585,10 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
         }
 
         let tcx = self.tcx;
+        let reachable_set = tcx.reachable_set(());
 
         let keys_and_jobs = tcx.mir_keys(()).iter().filter_map(|&def_id| {
```

Comment:
Should we filter more aggressively using a `.filter(|def_id| reachable_set.contains(def_id))`?

Reply:
That's my question: should we filter MIR for CTFE too according to the same rule?

Reply:
Could you add a line explaining the logic around `reachable_set` and this exception to the doc-comment on `should_encode_mir`?
r=me then